doc/kernel/other/memory_domain.rst

```diff
@@ -0,0 +1,176 @@
+a memory domain.
+
+.. code-block:: c
+
+    /* the start address of the MPU region needs to align with its size */
+    u8_t __aligned(32) app0_buf[32];
+    u8_t __aligned(32) app1_buf[32];
+
+    K_MEM_PARTITION_DEFINE(app0_part0, app0_buf, sizeof(app0_buf),
+                           K_MEM_PARTITION_P_RW_U_RW);
+
+    K_MEM_PARTITION_DEFINE(app0_part1, app1_buf, sizeof(app1_buf),
+                           K_MEM_PARTITION_P_RW_U_RO);
+
+    struct k_mem_partition *app0_parts[] = {
+        app0_part0,
+        app0_part1
+    };
+
+    k_mem_domain_init(&app0_domain, ARRAY_SIZE(app0_parts), app0_parts);
+
+This second code sample shows how to add memory partitions into a initialized
+memory domain one by one.
+
+.. code-block:: c
+
+    /* the start address of the MPU region needs to align with its size */
+    u8_t __aligned(32) app0_buf[32];
+    u8_t __aligned(32) app1_buf[32];
+
+    K_MEM_PARTITION_DEFINE(app0_part0, app0_buf, sizeof(app0_buf),
+                           K_MEM_PARTITION_P_RW_U_RW);
+
+    K_MEM_PARTITION_DEFINE(app0_part1, app1_buf, sizeof(app1_buf),
+                           K_MEM_PARTITION_P_RW_U_RO);
+
+    k_mem_domain_add_partition(&app0_domain, &app0_part0);
+    k_mem_domain_add_partition(&app0_domain, &app0_part1);
+
+.. note::
+    The maximum number of memory partitions is limited by the maximum
+    number of MPU regions or the maximum number of MMU tables.
+
+Add Threads into a Memory Domain
+================================
+
+Adding threads into a memory domain grants threads permission to access
+the memory partitions in the memory domain.
+
+The following code shows how to add threads into a memory domain.
+
+.. code-block:: c
+
+    k_mem_domain_add_thread(&app0_domain, &app_thread_id);
+
+Remove a Memory Partition from a Memory Domain
+==============================================
+
+The following code shows how to remove a memory partition from a memory
+domain.
+
+.. code-block:: c
+
+    k_mem_domain_remove_partition(&app0_domain, &app0_part1);
+
+The k_mem_domain_remove_partition() API would find the memory partition
+that matches the given parameter and remove it from the memory domain.
+
+Remove a Thread from the Memory Domain
+======================================
+
+The following code shows how to remove a thread from the memory domain.
+
+.. code-block:: c
+
+    k_mem_domain_remove_thread(&app0_domain, &app_thread_id);
+
+Destroy a Memory Domain
+=======================
+
+The following code shows how to destroy a memory domain.
+
+.. code-block:: c
+
+    k_mem_domain_destroy(&app0_domain);
+
+Available Partition Attributes
+==============================
+
+When defining a partition, we need to set access permission attributes
+to the partition. Since the access control of memory partitions relies on
+either an MPU or MMU, the available partition attributes would be architecture
+dependent.
+
+The following shows some of the available partition attributes, the complete
```

Comment:
Better would be:

     The complete list of available partition attributes for a specific architecture
     are found in the architecture-specific include file ``include/arch/<arch name>/arch.h``,
     (for example, ``include/arch/arm/arch.h``).  Some examples of partition attributes are:

Reply:
revised.